src/server/utils/server_diagnostics.js

```diff
@@ -17,120 +16,165 @@ const TMP_WORK_DIR = '/tmp/diag';
 const LONG_EXEC_TIMEOUT = 60 * 1000;
 
+// operations for diagnostics that can take place in parallel
+const operations = [
+    async () => {
+        try {
+            await collect_supervisor_logs();
+            return diag_log('collected supervisor logs successfully');
+        } catch (err) {
+            return diag_log('collect_supervisor_logs failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await copy_files('/log/noobaa_deploy*', TMP_WORK_DIR, 'fp');
+            return diag_log('collected noobaa_deploy logs successfully');
+        } catch (err) {
+            return diag_log('collecting noobaa_deploy.log failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await copy_files('/log/noobaa.log*', TMP_WORK_DIR, 'fp');
+            return diag_log('collected noobaa.log files successfully');
+        } catch (err) {
+            return diag_log('collecting noobaa.log failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await copy_files('/log/client_noobaa.log*', TMP_WORK_DIR, 'fp');
+            return diag_log('collected client_noobaa.log files successfully');
+        } catch (err) {
+            return diag_log('collecting client_noobaa.log failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await copy_files('/log/nbfedump', TMP_WORK_DIR, 'fpR');
+            return diag_log('collected /log/nbfedump directory successfully');
+        } catch (err) {
+            return diag_log('collecting /log/nbfedump directory failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await promise_utils.exec(`lsof &> ${TMP_WORK_DIR}/lsof.out`, {
+                ignore_rc: false,
+                return_stdout: false,
+                timeout: LONG_EXEC_TIMEOUT
+            });
+            return diag_log('collected lsof.out successfully');
+        } catch (err) {
+            return diag_log('collecting lsof.out failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await promise_utils.exec(`chkconfig &> ${TMP_WORK_DIR}'/chkconfig.out`, {
+                ignore_rc: false,
+                return_stdout: false,
+                timeout: LONG_EXEC_TIMEOUT
+            });
+            return diag_log('collected chkconfig.out successfully');
+        } catch (err) {
+            return diag_log('collecting chkconfig.out failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await copy_files('/var/log/messages*', TMP_WORK_DIR, 'fp');
+            return diag_log('collected /var/log/messages files successfully');
+        } catch (err) {
+            return diag_log('collecting /etc/noobaa files failed with error: ' + err);
+        }
+    },
+
+    async () => {
+        try {
+            await promise_utils.exec(`df -h &> ${TMP_WORK_DIR}/df.out`, {
```

Comment:
df is file system disk space usage, do we have any value in running it in a container?

Reply:
@liranmauda probably not a lot of value, but if it's not failing let's keep it for now